indico/modules/rb_new/client/js/components/SearchBar.jsx

```diff
@@ -38,157 +37,35 @@ export default class SearchBar extends React.Component {
     constructor(props) {
         super(props);
 
-        const {filters: {building, floor, text}} = this.props;
+        const {filters: {text}} = this.props;
         this.state = {
-            filtersPopupVisible: false,
-            filtersChanged: false,
-            inputTextValue: null,
-            roomName: text,
-            building,
-            floor
+            inputTextValue: text,
         };
     }
 
-    componentDidMount = () => {
-        this.setState({inputTextValue: this.composeInputTextValue()});
-    };
-
     componentWillUnmount = () => {
-        this.setState({building: null, floor: null, roomName: null, inputTextValue: null});
-    };
-
-    toggleFiltersPopup = (visible) => {
-        this.setState({filtersPopupVisible: visible});
-    };
-
-    close = () => {
-        const {filtersChanged} = this.state;
-        this.toggleFiltersPopup(false);
-
-        if (filtersChanged) {
-            this.setState({filtersChanged: false});
-        }
-    };
-
-    composeInputTextValue = () => {
-        const stateToKeys = {building: 'bldg', floor: 'floor'};
-        const {roomName} = this.state;
-
-        let textParts = Object.entries(stateToKeys)
-            .filter(([stateKey]) => {
-                return !!this.state[stateKey]; // eslint-disable-line react/destructuring-assignment
-            })
-            .map(([stateKey, searchKey]) => {
-                return `${searchKey}:${this.state[stateKey]}`; // eslint-disable-line react/destructuring-assignment
-            });
-
-        if (roomName) {
-            textParts = [roomName, ...textParts];
-        }
-
-        return textParts.length ? textParts.join(' ') : null;
-    };
-
-    didFiltersChange = () => {
-        const {filters} = this.props;
-        const {building, floor, roomName} = this.state;
-
-        return roomName !== filters.text || building !== filters.building || floor !== filters.floor;
+        this.setState({inputTextValue: null});
```

Comment:
Do we even need this? State is lost when a component is unmounted.

Reply:
Good catch, we don't